src/components/VmFilters/index.js

```diff
@@ -0,0 +1,207 @@
+      },
+    ]
+    return filterTypes
+  }
+
+  filterAdded (field, value) {
+    let activeFilters = Object.assign({}, this.props.vms.get('filters').toJS())
+    if ((field.filterType === 'select')) {
+      activeFilters[field.id] = value.id
+    } else {
+      if (!activeFilters[field.id]) {
+        activeFilters[field.id] = []
+      }
+      activeFilters[field.id].push(value)
+    }
+    this.props.onFilterUpdate(activeFilters)
+  };
+
+  selectFilterType (filterType) {
+    const { currentFilterType } = this.state
+    if (currentFilterType !== filterType) {
+      let newCurrentValue = ''
+      if (filterType.filterType === 'select') {
+        if (this.filterExists(filterType.id)) {
+          const filterValue = this.getFilterValue(filterType.id)
+          newCurrentValue = filterValue
+        }
+      }
+      this.setState({
+        currentFilterType: filterType,
+        currentValue: newCurrentValue,
+      })
+    }
+  }
+
+  filterValueSelected (filterValue) {
+    const { currentFilterType, currentValue } = this.state
+
+    if (filterValue !== currentValue) {
+      this.setState({ currentValue: filterValue })
+      if (filterValue) {
+        this.filterAdded(currentFilterType, filterValue)
+      }
+    }
+  }
+
+  updateCurrentValue (event) {
+    this.setState({ currentValue: event.target.value })
+  }
+
+  onValueKeyPress (keyEvent) {
+    const { currentValue, currentFilterType } = this.state
+
+    if (keyEvent.key === 'Enter' && currentValue && currentValue.length > 0) {
+      this.setState({ currentValue: '' })
+      this.filterAdded(currentFilterType, currentValue)
+      keyEvent.stopPropagation()
+      keyEvent.preventDefault()
+    }
+  }
+
+  filterExists (fieldId) {
+    return !!this.props.vms.getIn(['filters', fieldId])
+  };
+
+  getFilterValue (fieldId) {
+    return this.props.vms.getIn(['filters', fieldId])
+  };
+
+  renderInput () {
+    const { currentFilterType, currentValue, filterCategory } = this.state
+    if (!currentFilterType) {
+      return null
+    }
+
+    if (currentFilterType.filterType === 'select') {
+      if (currentValue !== '' && !this.filterExists(currentFilterType.id)) {
+        this.setState({
+          currentValue: '',
+          filterCategory,
+        })
+      }
+      return (
+        <Filter.ValueSelector
+          filterValues={currentFilterType.filterValues}
+          placeholder={currentFilterType.placeholder}
+          currentValue={currentValue}
+          onFilterValueSelected={this.filterValueSelected}
+        />
+      )
+    }
+    return (
+      <FormControl
+        type={currentFilterType.filterType}
+        value={currentValue}
+        placeholder={currentFilterType.placeholder}
+        onChange={e => this.updateCurrentValue(e)}
+        onKeyPress={e => this.onValueKeyPress(e)}
+      />
+    )
+  }
+
+  componentDidUpdate () {
+    saveToLocalStorage('vmFilters', JSON.stringify(this.props.vms.get('filters').toJS()))
```

Comment:
I think this will have the problem of the filters / sorts being stored for the app and not a particular user of the app.  Login with "userA", set a filter, logout, login with "userB" and the original filter would still be applied.